app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -265,6 +267,7 @@ class LocationSensorManager : LocationSensorManagerBase() {
                     } else {
                         Log.d(TAG, "High accuracy mode parameters changed. Disable high accuracy mode.")
                         stopHighAccuracyService()
+                        delay(500)
```

Comment:
Do you know what exactly caused the issue? Because location updates are received through intents, if I had to guess maybe it's a race condition with the `PendingIntent` we create so perhaps the foreground service intent shouldn't use the same ID as the background service intent to make sure they don't influence each other / cancel out? Both services request `removeLocationUpdates` with (usually) basically the same intent.

I'm unable to reproduce the bug so I can't test it, but simply adding a delay feels like a bad hack and should be avoided.

Reply:
> Do you know what exactly caused the issue?

I think it is a speed issue, not entirely sure if its caused by the other changes in this PR moving away from deprecated methods?



> Because location updates are received through intents, if I had to guess maybe it's a race condition with the `PendingIntent` we create so perhaps the foreground service intent shouldn't use the same ID as the background service intent to make sure they don't influence each other / cancel out? Both services call `requestLocationUpdates` and `removeLocationUpdates` with (usually) the same intent.

That is a good point, let me try and test that out here, woudl be a quick change to use a different intent that hits the same function.



> I'm unable to reproduce the bug so I can't test it, but simply adding a delay feels like a bad hack and should be avoided.

I agree, I wasn't really sure how to proceed here but when I looked at my logs it was definitely an issue. With this delay I continue to see updates flowing in every minute but without it I don't see updates coming so it could be the intent conflict you mentioned.

Reply:
working well without a delay, as always thank you @jpelgrom 🙏

Reply:
I was actually referring to the `requestCode` (currently `0`), but this also works to make the intents unique!

Reply:
Oh hmm I don't see `requestCode` in either of the location files lol do you think we should still change it given I no longer see the delayed issue?

Reply:
`requestCode` is the [second parameter for `PendingIntent.getBroadcast`](https://developer.android.com/reference/kotlin/android/app/PendingIntent?hl=en#getbroadcast)

No need to change it now, it works and might come in handy in the future to be able to handle high accuracy updates differently.